### DISCUSSION TITLE
How to play sound from assets folder?

### DISCUSSION THREAD

**Discussion by vimalmistry** [bluefireteam/audioplayers#965] [Q&A]
How to play .mp3 and .wav files from assets folder. I tried with is local but its shows file not found. So How it works? I read all document. I couldn't found any solution. Please any one?  I want to play just 2-3 sound which are fixed.

**Comment by erickzanardo** [comment]
You need to use AudioCache to play assets files

**Reply by sahilmeta1** [reply]
I am a beginner, read all the documentation-- it has 3 lines of decription for audiocache but not proper code to use it 
like player.play('filename') and when i am using the method its showing error

**Reply by Gustl22** [reply]
It's handled in Audioplayer now, no need for AudioCache since 1.x.x see [example](https://github.com/bluefireteam/audioplayers/blob/3ee12cd0361c0fc2f3d0303c504732d12fa8e49a/packages/audioplayers/example/lib/tabs/sources.dart#L94)

**Comment by mohammaddanishcoder** [comment]
You can Play Sound from Assets Folder by instantiating AudioPlayer and then using this line of code
player.play(AssetSource(('audio/assets_note1.wav')));
Below is the sample code:
 TextButton(
            onPressed: ()  {
              AudioPlayer player=AudioPlayer();
              player.play(AssetSource(('audio/assets_note1.wav')));
            },
            child: const Text('Click Me'),
          ),

**Comment by Jkanuta** [comment]
The audiocache method is deprecated now. It's no longer supported. We now use AudioPlayer() for this functionality. AudioPlayer detects sound automatically in a /assets folder.

Just import =>

> import 'package:audioplayers/audioplayers.dart';

Then go to your pubspec.yaml file and add the dependency.

Say you are using a TextButton for action, write this on your main.dart file =>

`TextButton(
              onPressed: () {
                AudioPlayer player = AudioPlayer();
                player.play(
                  AssetSource('your_song.wav'),
                );
              },
              child: Text('Click me'),
            )`

**Comment by jorgefuertes** [comment]
None of this works for me. `Audioplayer.play` seems to be a Future, and it raises an `asynchronous suspension`.

I'm new to Dart and may I am wrong, but documentation seems outdated for this package.

Weird, it worked after a clean rebuild.

~~~dart
void playNote1() async {
  final player = AudioPlayer();
  await player.play(AssetSource("note1.wav"));
}
~~~
